app/src/main/java/de/tum/in/tumcampusapp/component/ui/ticket/adapter/EventsAdapter.java

```diff
@@ -0,0 +1,128 @@
+
+        public EventViewHolder(View view) {
+            super(view);
+            cardView = (CardView) view;
+            titleView = view.findViewById(R.id.events_title);
+            imgView = view.findViewById(R.id.events_img);
+            localityView = view.findViewById(R.id.events_src_locality);
+            srcDateView = view.findViewById(R.id.events_src_date);
+        }
+    }
+
+    public EventsAdapter(List<Event> events) {
+        mEventList = events;
+    }
+
+    @Override
+    public EventViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
+        if (mContext == null) {
+            mContext = parent.getContext();
+        }
+        View view = LayoutInflater.from(mContext)
+                .inflate(R.layout.card_events_item, parent, false);
+        return new EventViewHolder(view);
+    }
+
+    public static void bindNewsView(RecyclerView.ViewHolder newsViewHolder, Event event, Context context) {
+        EventViewHolder holder = (EventViewHolder) newsViewHolder;
+
+        holder.imgView.setVisibility(View.VISIBLE);
+        holder.titleView.setVisibility(View.VISIBLE);
+
+        // Set image
+        String imgUrl = event.getImage();
+        if (imgUrl == null || imgUrl.isEmpty()) {
+            if (event.getLink().endsWith(".png") || event.getLink().endsWith(".jpeg")) {
+                Utils.log("try link as image");
+                // the link points to an image (eventspread)
+                Picasso.get()
+                        .load(event.getLink())
+                        .placeholder(R.drawable.chat_background)
+                        .into(holder.imgView, new Callback() {
+                            @Override
+                            public void onSuccess() {
+                                holder.titleView.setVisibility(View.GONE); // titleView is included in eventspread slide
+                                holder.imgView.setOnClickListener(null); // link doesn't lead to more infos
+                            }
+
+                            @Override
+                            public void onError(Exception e) {
+                                holder.imgView.setVisibility(View.GONE); // we can't display the image after all
+                            }
+                        });
+            } else {
+                holder.imgView.setVisibility(View.GONE);
+            }
+        } else {
+            Picasso.get()
+                    .load(imgUrl)
+                    .placeholder(R.drawable.chat_background)
+                    .into(holder.imgView);
+        }
+
+        String title = event.getTitle();
+        title = COMPILE.matcher(title)
+                .replaceAll("");
+        holder.titleView.setText(title);
+
+        //Adds localityView
+        String locality = event.getLocality();
+        holder.localityView.setText(locality);
+
+        // Adds date
+        holder.srcDateView.setText(Event.Companion.getFormattedDate(event.getStart()));
```

Comment:
Since we do not only use the formatting methods in this class for the event start date, but also to format other dates with the same pattern, I would keep it as a method.